src/Pages/UserAccount/PaymentMethodPage.js

```diff
@@ -0,0 +1,325 @@
+import React, { useEffect, useState } from "react";
+import Modal from "./Components/PaymentModal";
+import { Field, Form, Formik } from "formik";
+import CustomTextInput from "../../Components/TextInput";
+import { PaymentSchema } from "../../ValidationSchemas/PaymentSchema";
+import { useQuery } from "react-query";
+import UseAnimation from "../../Components/Loader";
+import loading from "react-useanimations/lib/loading";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth, database } from "../../FirebaseConfigs/Firesbase";
+import { PaymentMethodServices } from "../../Services/AccountServices";
+import { push, set, ref } from "firebase/database";
+import PaymentCardItem from "./Components/PaymentCardItem";
+import ActionButton from "./Components/ActionButton";
+import { closeOutline } from "ionicons/icons";
+
+// TODO: FIX EDIT AND DELETE PAYMENT HANDLERS.
+
+export default function PaymentMethodPage() {
+  const [paymentModal, setPaymentModal] = useState(false);
+  const [editModal, setEditModal] = useState(false);
+  const [payer, setPayer] = useState(null);
+  const [payerId, setPayerId] = useState(null);
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [mobilePayment, setMobilePayment] = useState(false);
+  console.log("paymentUi: ", mobilePayment);
+
+  const {
+    data = [],
+    isLoading,
+    refetch,
+    isError,
+    error,
+  } = useQuery(["bankCard", payerId], () => PaymentMethodServices(payerId));
+  console.log("Bank card: ", data);
+  console.log("payerId:", payerId);
+
+  useEffect(() => {
+    const subscribed = onAuthStateChanged(auth, (data) => {
+      if (data) {
+        setPayer(data.displayName);
+        setPayerId(data.uid);
+        const [first, last] = data.displayName.split(" ");
+        setFirstName(first);
+        setLastName(last);
+      } else {
+        setPayer(null);
+      }
+    });
+    return () => {
+      subscribed();
+    };
+  }, []);
+
+  function modalHandler() {
+    setPaymentModal(!paymentModal);
+  }
+
+  const editPaymentHandler = () => {
```

Comment:
Firgure out how to retrieve data for a single payment method and proceed to implementing editHandler